client/src/main/scala/sbtcompendium/client/CompendiumClient.scala

```diff
@@ -102,7 +106,17 @@ object CompendiumClient {
         } yield out
       }
 
-      override def generateClient(target: IdlName, identifier: String): F[String] = F.pure("")
+      override def generateClient(target: IdlName, identifier: String): F[List[String]] =
+        target match {
+          case IdlName.Avro =>
+            retrieveProtocol(identifier, None)
+              .map(_.map(r => handleAvro(r.raw)).getOrElse(List.empty))
+          case _ =>
+            Sync[F].raiseError(UnknownError(s"Unknown error with status code 501. Schema format not implemented yet"))
+        }
+
+      private def handleAvro(raw: String): List[String] =
+        Generator(Standard).stringToStrings(raw)
```

Comment:
This can be considered an _effect_ that we might want to encapsulate in the F context.

Let me suggest you the following:

```scala
override def generateClient(target: IdlName, identifier: String): F[List[String]] =
        target match {
          case IdlName.Avro =>
            for {
              protocol <- retrieveProtocol(identifier, None)
              code     <- handleAvro(protocol)
            } yield code

          case _ =>
            F.raiseError(
              UnknownError(
                s"Unknown error with status code 501. " +
                  s"Schema format not implemented yet"
              )
            )
        }

      private def handleAvro(protocol: Option[Protocol]): F[List[String]] =
        protocol.map(r => F.delay(Generator(Standard).stringToStrings(r.raw))).getOrElse(F.pure(Nil))
```

Reply:
agree, didn't realise